### ISSUE TITLE
[java] PreserveStackTrace false-positive

### ISSUE THREAD

**Issue by boris-petrov** [pmd/pmd#1747]
PMD version 6.13.0.

Rule: [PreserveStackTrace](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#preservestacktrace)

```java
try {
	...
} catch (IOException exception) {
	if (exception.getMessage().contains("foobar")) {
		throw new SomeOtherUserDefinedException("Error", exception);
	}
	throw exception;
}
```

This code produces a `PreserveStackTrace` error but it shouldn't according to the documentation of `PreserveStackTrace`.

**Comment by adangel** [comment]
This has been fixed with PMD 7.0.0-rc1.
